### PR TITLE
To verify readOnlyCells

### PR DESCRIPTION
**Pull request by Nehani**
To verify readOnly

### CODE REVIEW

File: Keywords/commonUtils/WebTableUtils.groovy
Comment (smitagarwal1061) [inline Keywords/commonUtils/WebTableUtils.groovy:1]: - why you PR shows you deleted all the existing code from the class and then added again?
- how come your PR does not have any updated cache files? Did you miss committing them?

File: Keywords/commonUtils/WebTableUtils.groovy [hunk 1]
```diff
@@ -1,253 +1,285 @@
...
+			} else {
+				KeywordUtil.markFailedAndStop("Cell with row name '" + rowName + "' and column name '" + columnName + "' not found in the table with primarykey: '"+primaryKey+"'")
+			}
+			return cellValue;
+		}
+		catch (Exception e) {
+			WebUI.comment("Exception :: " + e.getMessage())
+			KeywordUtil.markFailedAndStop("Exception while fetching cell value from WebTable");
+		}
+	}
```
Comment (smitagarwal1061) [inline Keywords/commonUtils/WebTableUtils.groovy:254]: line break before starting a new method is missing

File: Keywords/commonUtils/WebTableUtils.groovy [hunk 2]
```diff
@@ -1,253 +1,285 @@
...
+	/**
+	 * to get the read only cells
```
Comment (smitagarwal1061) [inline Keywords/commonUtils/WebTableUtils.groovy]: - Inappropriate description, do we fetch Read only cells in this keyword?
It should be as below
Verify all cells of a web table are in read only mode i.e. disabled for editing

File: Keywords/commonUtils/WebTableUtils.groovy [hunk 3]
```diff
@@ -1,253 +1,285 @@
...
+	 * @param tableTitle - Title of the table to locate
+	 * @param disabled   - The class that indicates a readOnly state
```
Comment (smitagarwal1061) [inline Keywords/commonUtils/WebTableUtils.groovy]: name given to argument is not meaningful. It should be
attributeClassValue - value of class for a disabled cell in web table

File: Keywords/commonUtils/WebTableUtils.groovy [hunk 4]
```diff
@@ -1,253 +1,285 @@
...
+	 *
+	 */
+	@Keyword(keywordObject = 'readOnly')
```
Comment (smitagarwal1061) [inline Keywords/commonUtils/WebTableUtils.groovy]: Did we not discuss in morning? Are you still confused?

File: Keywords/commonUtils/WebTableUtils.groovy [hunk 5]
```diff
@@ -1,253 +1,285 @@
...
+	public static void verifyTableCellIsEnabled(String tableTitle, String disabled) {
```
Comment (smitagarwal1061) [inline Keywords/commonUtils/WebTableUtils.groovy]: We are validating that webtable cell is disabled or not but your function name is "verifyTableCellIsEnabled". Is it not contradictory?

File: Keywords/commonUtils/WebTableUtils.groovy [hunk 6]
```diff
@@ -1,253 +1,285 @@
...
+				// Iterate through cells in each row
+				for (int i = 0; i < allCells.size(); i++) {
+					WebElement cell = allCells.get(i);
+					String cellClass = cell.getAttribute("class");
+					if (cellClass != null) {
+						if (cellClass.equals("disabled")) {
+							KeywordUtil.logInfo("Cell is in readonly mode with value " + cell.getText());
+						} else {
+							KeywordUtil.markFailedAndStop("Cell is enabled mode with value " + cell.getText());
+						}
```
Comment (smitagarwal1061) [inline Keywords/commonUtils/WebTableUtils.groovy]: No need to print cell value for pass condition. If there are 100 cells, report will become unreadable
Just log failure and stop if condition is not met.

**Review by smitagarwal1061 — CHANGES_REQUESTED** [review]

File: Keywords/commonUtils/WebTableUtils.groovy (hunk 6 above)
Comment (Nehani) [reply Keywords/commonUtils/WebTableUtils.groovy]: Done

File: Keywords/commonUtils/WebTableUtils.groovy (hunk 4 above)
Comment (Nehani) [reply Keywords/commonUtils/WebTableUtils.groovy]: Done

File: Keywords/commonUtils/WebTableUtils.groovy
Comment (Nehani) [reply Keywords/commonUtils/WebTableUtils.groovy:1]: Not sure why, I just added my at the last, nor I changed anything in previous code

File: Keywords/commonUtils/WebTableUtils.groovy (hunk 2 above)
Comment (Nehani) [reply Keywords/commonUtils/WebTableUtils.groovy]: Done

File: Keywords/commonUtils/WebTableUtils.groovy (hunk 1 above)
Comment (Nehani) [reply Keywords/commonUtils/WebTableUtils.groovy:254]: Done

File: Keywords/commonUtils/WebTableUtils.groovy (hunk 3 above)
Comment (Nehani) [reply Keywords/commonUtils/WebTableUtils.groovy]: Done

File: Keywords/commonUtils/WebTableUtils.groovy (hunk 5 above)
Comment (Nehani) [reply Keywords/commonUtils/WebTableUtils.groovy]: Done

**Review by smitagarwal1061 — APPROVED** [review]

**Pull request merged**